### ISSUE TITLE
Opacity broken for VBO-based batched Highlight, Select and X-Ray rendering - shows as opaque white

### ISSUE THREAD

**Issue by xeolabs** [xeokit/xeokit-sdk#1335]
Reproduce with this example: https://xeokit.github.io/xeokit-sdk/examples/scenemodel/#effects_vbo_batching_xray

Only affects batched geometries in VBO scene representations. Likely an RGBA alpha channel not being uploaded or rendered correctly in a shader. 

WebGL blending is being switched correctly.

What I've tried:

* Forcing gl.BLEND on all the time
* Ensuring that only the Xray shader is the one drawing
* Different alpha blending functions
* Forcing outgoing fragment to vec4(1.0, 0.0, 0.0, 0.5)
* Disabling picking to ensure no interference from other shaders

Still renders a solid white surface. No idea why.

````javascript
    const viewer = new Viewer({
        canvasId: "myCanvas",
        transparent: true,
        dtxEnabled: false // <<---------- Use VBOs
    });

    viewer.scene.xrayMaterial.fillColor = [1.0, 0.0, 0.0];
    viewer.scene.xrayMaterial.fillAlpha = 0.5;

    const xktLoader = new XKTLoaderPlugin(viewer);

    const sceneModel = xktLoader.load({
        id: "myModel",
        src: "../../assets/models/xkt/v10/glTF-Embedded/Duplex_A_20110505.glTFEmbedded.xkt",
        edges: true
    });
````

![Screenshot from 2024-01-18 11-31-39](https://github.com/xeokit/xeokit-sdk/assets/83100/514e293e-a5b4-4faa-a648-1b8376e712b0)

````javascript
    const viewer = new Viewer({
        canvasId: "myCanvas",
        transparent: true,
        dtxEnabled: true // <<---------- Use data textures
    });

    viewer.scene.xrayMaterial.fillColor = [1.0, 0.0, 0.0];
    viewer.scene.xrayMaterial.fillAlpha = 0.5;

    const xktLoader = new XKTLoaderPlugin(viewer);

    const sceneModel = xktLoader.load({
        id: "myModel",
        src: "../../assets/models/xkt/v10/glTF-Embedded/Duplex_A_20110505.glTFEmbedded.xkt",
        edges: true
    });
````

![Screenshot from 2024-01-18 11-37-06](https://github.com/xeokit/xeokit-sdk/assets/83100/c87f1307-342d-4133-bab9-4800dd7c7205)

SceneModel built using batched geometries and VBO-based scene representation and rendering:

![Screenshot from 2024-01-18 12-04-01](https://github.com/xeokit/xeokit-sdk/assets/83100/444ed397-8b02-4465-976b-b48e06d706c0)

SceneModel built using instanced geometries and VBO-based scene representation and rendering:

![Screenshot from 2024-01-18 12-05-38](https://github.com/xeokit/xeokit-sdk/assets/83100/a1f235ef-3dc2-46fd-9037-019d51cca2f2)

**Comment by xeolabs** [comment]
@Kurtil I'm a bit stumped on this one and it may have something to do with the shader refactor, would you be able to take a look? 

Note that I renamed the various VBO renderer classes and refactored them out into their own files, but AFAIK I haven't changed anything else that would break this. 

One thing that seems odd is that in VBORenderer, the base class for all VBO renderers, if I substitute a value of `[1,0,0,0.5]`for `defaultColor` on line 626 the white color (incorrectly) applied for X-Ray, Highlight and Selection becomes that color.

This is strange, because that code block should be for objects that are not highlighted/selected/xrayed.

**Comment by xeolabs** [comment]
Fixed in v2.5.2-beta-13
